src/plugins/intel_cpu/src/nodes/reduce.cpp

```diff
@@ -2274,6 +2274,14 @@ void Reduce::execute(dnnl::stream strm) {
     const uint8_t *src_data = srcMemPtr->getDataAs<const uint8_t>();
     uint8_t *dst_data = dstMemPtr->getDataAs<uint8_t>();
 
+    const auto& src_shape = srcMemPtr->getStaticDims();
```

Comment:
@dmitry-gorokhov I agree. If the expected output shape is non-empty, then we should apply post ops fusion.
@mitruska  Could you please add `reduce_kernel_post_process(dst_data);` before the new added early `return`?

Reply:
An alternative way would be to disable post ops fusion [here](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/reduce.cpp#L3428) if input is empty, which would be more safe. After all this is a corner case to reduce dim size 0 to 1, any incompatibility to the variants of such case from the computation module would lead to incorrect/undefined results.

Reply:
> An alternative way would be to disable post ops fusion [here](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/reduce.cpp#L3428) if input is empty, which would be more safe. After all this is a corner case to reduce dim size 0 to 1, any incompatibility to the variants of such case from the computation module would lead to incorrect/undefined results.

@xuchen-intel Shapes might be dynamic. So we don't know in advanced should the fusion be disabled or not.

Reply:
Right! Then @mitruska please let's add `reduce_kernel_post_process(dst_data);`, with some test cases (e.g. [here](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/instances/x64/reduce.cpp)) to reproduce this issue beforehand if possible.
Applying `reduce_kernel_post_process(dst_data);` should work for post ops cases. Otherwise, please let me know, if you need me to fix it in a separate PR for the CPU part. @dmitry-gorokhov

Reply:
@dmitry-gorokhov, @xuchen-intel Thank you for your support! I've added the `reduce_kernel_post_process(dst_data)`, but it needed some special conditions to make the tests passed. 

The condition:
https://github.com/openvinotoolkit/openvino/blob/99c1e89c90d9244c1e40a05ebc779d0a27f21aaf/src/plugins/intel_cpu/src/nodes/reduce.cpp#L2281-L2283

was inspired by:
https://github.com/openvinotoolkit/openvino/blob/99c1e89c90d9244c1e40a05ebc779d0a27f21aaf/src/plugins/intel_cpu/src/nodes/reduce.cpp#L2564-L2565

Moreover I was able to use only fuse "Swish" CPU tests, because other fusion tests pipeline was failing for 0 and dynamic cases. Those cases probably need some fusing test pipeline adjustment.